web/html/src/manager/notifications/notifications-list.tsx

```diff
@@ -0,0 +1,454 @@
+import * as React from "react";
```

Comment:
Would be nice to chop this file up a little bit, right now it's starting to be a bit of a mouthful. Maybe slice by types, comparators, etc, but I don't have a strong preference here.

Reply:
I extracted the types and the comparators into separe typescript source file. For the comparators I used a static class, I can change it to an object literal and functions if that's a preferred implementation style.

Reply:
That looks good, no strong preference on which way to go on that front. 👍